warp10/src/main/java/io/warp10/script/filter/FilterByLabels.java

```diff
@@ -136,12 +136,12 @@ public FilterByLabels(String name, Map<String,String> selectors, boolean checkLa
           }
 
           // Now check the label or attribute value
-          if (hasLabel && checkLabels) {
+          if (hasLabel) {
             if (!matcher.reset(labels.get(label)).matches()) {
               matched = false;
               break;
             }            
-          } else if (checkAttributes) {
```

Comment:
Because
```
          if (!matched) {	          if (!matched) {
            break;	            break;
          }
```
We know that it matched and it was not on labels, so it must be on attributes.

Reply:
Maybe, but should we check the attributes? This is what checkAttributes determines, so it is not always true.

Reply:
In the else we have `!hasLabel && matched` so `matched != hasLabel`. This means we were not in the 
```
else {
            matched = hasLabel;
          }
```
branch but in the 
```
          if (!hasLabel && checkAttributes) {
            if (!attributes.containsKey(label)) {
              matched = false;
            }
          }
```
branch.

Which means `checkAttributes == true`.

Reply:
Then add it as a comment to make the understanding easier once the clause is removed.